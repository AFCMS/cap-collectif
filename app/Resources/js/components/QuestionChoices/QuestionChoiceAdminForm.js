// @flow
import * as React from 'react';
import { connect, type MapStateToProps } from 'react-redux';
import { formValueSelector, arrayPush } from 'redux-form';
import { FormattedMessage, injectIntl, type IntlShape } from 'react-intl';
import { ListGroup, ListGroupItem, ButtonToolbar, Button, Row, Col } from 'react-bootstrap';
import QuestionChoiceAdminModal from './QuestionChoiceAdminModal';
import type { GlobalState, Dispatch } from '../../types';

type QuestionChoices = $ReadOnlyArray<{|
  +id: string,
  +title: string,
  +description: ?string,
  +color: ?string,
  +image: ?Object,
|}>;

type Props = {
  dispatch: Dispatch,
  fields: { length: number, map: Function, remove: Function },
  choices: QuestionChoices,
  formName: string,
  oldMember: string,
  type: string,
  intl: IntlShape,
};

type State = { editIndex: ?number };

export class QuestionChoiceAdminForm extends React.Component<Props, State> {
  state = {
    editIndex: null,
  };

  handleClose = (index: number) => {
    const { fields, choices } = this.props;
    if (!choices[index].id) {
      fields.remove(index);
    }
    this.handleSubmit();
  };

  handleSubmit = () => {
    this.setState({ editIndex: null });
  };

  render() {
    const { dispatch, fields, choices, formName, oldMember, type, intl } = this.props;
    const { editIndex } = this.state;
    return (
      <div className="form-group" id="questions_choice_panel_personal">
        <ListGroup>
          {fields.map((member, index) => (
            <ListGroupItem key={index}>
              <QuestionChoiceAdminModal
                isCreating={!!choices[index].id}
                onClose={() => {
                  this.handleClose(index);
                }}
                onSubmit={this.handleSubmit}
                member={member}
                show={index === editIndex}
                formName={formName}
                type={type}
              />
              <Row>
                <Col xs={8}>
                  <div>
                    <strong>{choices[index].title}</strong>
                  </div>
                </Col>
                <Col xs={4}>
                  <ButtonToolbar className="pull-right">
                    <Button
                      bsStyle="warning"
                      className="btn--outline"
                      onClick={() => {
                        this.setState({ editIndex: index });
                      }}>
                      <i className="fa fa-pencil" /> <FormattedMessage id="global.edit" />
                    </Button>
                    <Button
                      bsStyle="danger"
                      className="btn--outline"
                      onClick={() => {
                        if (
                          window.confirm(
                            intl.formatMessage({ id: 'responses.alert.delete' }),
                            intl.formatMessage({ id: 'responses.alert.delete.bodyText' }),
                          )
                        ) {
                          fields.remove(index);
                        }
                      }}>
                      <i className="fa fa-trash" />
                    </Button>
                  </ButtonToolbar>
                </Col>
              </Row>
            </ListGroupItem>
          ))}
        </ListGroup>
        <Button
          bsStyle="primary"
          className="btn--outline box-content__toolbar"
          onClick={() => {
            dispatch(arrayPush(this.props.formName, `${oldMember}.choices`, {}));
            this.setState({ editIndex: fields.length });
          }}>
          <i className="fa fa-plus-circle" /> <FormattedMessage id="global.add" />
        </Button>
      </div>
    );
  }
}

const mapStateToProps: MapStateToProps<*, *, *> = (state: GlobalState, props: Props) => {
  const selector = formValueSelector(props.formName);
  return {
    choices: selector(state, `${props.oldMember}.choices`),
  };
};

export default connect(mapStateToProps)(injectIntl(QuestionChoiceAdminForm));
