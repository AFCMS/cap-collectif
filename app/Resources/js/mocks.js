// @flow
/* eslint-env jest */
import { type FormProps } from 'redux-form';
import { type IntlShape } from 'react-intl';
import type { RelayPaginationProp, RelayRefetchProp } from 'react-relay';

export const $refType: any = null;
export const $fragmentRefs: any = null;

export const intlMock: IntlShape = {
  locale: 'fr-FR',
  formats: {},
  messages: {},
  now: () => 0,
  formatHTMLMessage: message => String(message),
  formatPlural: message => String(message),
  formatNumber: message => String(message),
  formatRelative: message => String(message),
  formatTime: message => String(message),
  formatDate: message => String(message),
  formatMessage: message => String(message.id),
};

export const formMock: FormProps = {
  anyTouched: false,
  array: {
    insert: jest.fn(),
    move: jest.fn(),
    pop: jest.fn(),
    push: jest.fn(),
    remove: jest.fn(),
    removeAll: jest.fn(),
    shift: jest.fn(),
    splice: jest.fn(),
    swap: jest.fn(),
    unshift: jest.fn(),
  },
  asyncValidate: jest.fn(),
  asyncValidating: false,
  autofill: jest.fn(),
  blur: jest.fn(),
  change: jest.fn(),
  clearAsyncError: jest.fn(),
  clearSubmit: jest.fn(),
  destroy: jest.fn(),
  dirty: false,
  dispatch: jest.fn(),
  error: null,
  form: 'formName',
  handleSubmit: jest.fn(),
  initialize: jest.fn(),
  initialized: true,
  initialValues: {},
  invalid: false,
  pristine: true,
  reset: jest.fn(),
  submitting: false,
  submitFailed: false,
  submitSucceeded: false,
  touch: jest.fn(),
  untouch: jest.fn(),
  valid: true,
  warning: null,
};

export const relayPaginationMock: RelayPaginationProp = {
  environment: {
    applyMutation: jest.fn(),
    sendMutation: jest.fn(),
    lookup: jest.fn(),
    sendQuery: jest.fn(),
    subscribe: jest.fn(),
    streamQuery: jest.fn(),
    retain: jest.fn(),
    unstable_internal: {},
  },
  hasMore: () => false,
  isLoading: () => false,
  loadMore: jest.fn(),
  refetchConnection: jest.fn(),
};

export const relayRefetchMock: RelayRefetchProp = {
  environment: {
    applyMutation: jest.fn(),
    sendMutation: jest.fn(),
    lookup: jest.fn(),
    sendQuery: jest.fn(),
    subscribe: jest.fn(),
    streamQuery: jest.fn(),
    retain: jest.fn(),
    unstable_internal: {},
  },
  refetch: jest.fn(),
};
