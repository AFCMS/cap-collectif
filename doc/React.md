### How to write localizable components

Just import the appropriate [component](https://github.com/yahoo/react-intl/wiki#the-react-intl-module) from `react-intl`

- For localizable text use
[`<FormattedMessage>`](https://github.com/yahoo/react-intl/wiki/Components#formattedmessage).

- For date and time:
[`<FormattedDate>`](https://github.com/yahoo/react-intl/wiki/Components#formatteddate)
[`<FormattedTime>`](https://github.com/yahoo/react-intl/wiki/Components#formattedtime)
[`<FormattedRelative>`](https://github.com/yahoo/react-intl/wiki/Components#formattedrelative)

- For numbers and currencies:
[`<FormattedNumber>`](https://github.com/yahoo/react-intl/wiki/Components#formattednumber)
[`<FormattedPlural>`](https://github.com/yahoo/react-intl/wiki/Components#formattedplural)

- If possible, do not use `<FormattedHTMLMessage>`, see how to use *Rich Text Formatting* with
[`<FormattedMessage>`](https://github.com/yahoo/react-intl/wiki/Components#formattedmessage)

- When you need an imperative formatting API, use the [`injectIntl`](https://github.com/yahoo/react-intl/wiki/API#injectintl) High-Order Component.
