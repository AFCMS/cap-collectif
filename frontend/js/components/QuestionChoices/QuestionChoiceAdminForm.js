// @flow
import * as React from 'react';
import { connect } from 'react-redux';
import { arrayPush, change, formValueSelector } from 'redux-form';
import { FormattedMessage, injectIntl, type IntlShape } from 'react-intl';
// TODO https://github.com/cap-collectif/platform/issues/7774
// eslint-disable-next-line no-restricted-imports
import { Button, ButtonToolbar, Col, ListGroup, ListGroupItem, Row } from 'react-bootstrap';
import QuestionChoiceAdminModal from './QuestionChoiceAdminModal';
import type { Dispatch, GlobalState } from '~/types';
import QuestionnaireAdminModalImportResponses from '~/components/Questionnaire/QuestionnaireAdminModalImportResponses';

type QuestionChoices = $ReadOnlyArray<{|
  +id: string,
  +title: string,
  +description: ?string,
  +color: ?string,
  +image: ?Object,
|}>;

type Props = {|
  dispatch: Dispatch,
  fields: { length: number, map: Function, remove: Function },
  choices: QuestionChoices,
  formName: string,
  oldMember: string,
  type: string,
  intl: IntlShape,
  importedResponses: Object,
|};

type State = {|
  editIndex: number,
  editMember: ?string,
  showQuestionChoiceModal: boolean,
  showModal: boolean,
  showAll: boolean,
  isCreating: boolean,
|};

export class QuestionChoiceAdminForm extends React.Component<Props, State> {
  state = {
    editIndex: 0,
    editMember: null,
    showQuestionChoiceModal: false,
    showModal: false,
    showAll: false,
    isCreating: false,
  };

  handleClose = (index: number) => {
    const { fields, choices } = this.props;
    if (!choices[index].id) {
      fields.remove(index);
    }
    this.handleSubmit();
  };

  handleSubmit = () => {
    this.setState({ editIndex: 0, showQuestionChoiceModal: false });
  };

  render() {
    const {
      dispatch,
      fields,
      choices,
      formName,
      oldMember,
      type,
      intl,
      importedResponses,
    } = this.props;
    const {
      editIndex,
      showModal,
      showAll,
      editMember,
      showQuestionChoiceModal,
      isCreating,
    } = this.state;
    const MAX_FIELDS_DISPLAYED = 10;
    return (
      <div className="form-group" id="questions_choice_panel_personal">
        {showModal && (
          <QuestionnaireAdminModalImportResponses
            show={showModal}
            formName={formName}
            type={type}
            onClose={() => {
              this.setState({ showModal: false });
              dispatch(change(formName, `${oldMember}.importedResponses`, null));
            }}
            dispatch={dispatch}
            fields={fields}
            choices={choices}
            oldMember={oldMember}
            onAdd={() => {
              const responses = choices
                ? [...importedResponses.data, ...choices]
                : [...importedResponses.data];
              dispatch(change(formName, `${oldMember}.choices`, responses));
              dispatch(change(formName, `${oldMember}.importedResponses`, null));
              this.setState({ showModal: false });
            }}
          />
        )}
        <QuestionChoiceAdminModal
          isCreating={!isCreating}
          onClose={() => {
            this.handleClose(editIndex);
          }}
          onSubmit={this.handleSubmit}
          member={editMember}
          show={showQuestionChoiceModal}
          formName={formName}
          type={type}
        />
        <ListGroup>
          {fields.map((member, index) =>
            index < MAX_FIELDS_DISPLAYED || showAll ? (
              <ListGroupItem key={index}>
                <Row>
                  <Col xs={8}>
                    <div>
                      <strong>{choices[index].title}</strong>
                    </div>
                  </Col>
                  <Col xs={4}>
                    <ButtonToolbar className="pull-right">
                      <Button
                        id="edit-choice"
                        bsStyle="warning"
                        className="btn--outline"
                        onClick={() => {
                          this.setState({
                            editIndex: index,
                            editMember: member,
                            isCreating: false,
                            showQuestionChoiceModal: true,
                          });
                        }}>
                        <i className="fa fa-pencil" /> <FormattedMessage id="global.edit" />
                      </Button>
                      <Button
                        bsStyle="danger"
                        id="remove-choice"
                        className="btn--outline"
                        onClick={() => {
                          if (
                            window.confirm(
                              intl.formatMessage({ id: 'responses.alert.delete' }),
                              intl.formatMessage({ id: 'responses.alert.delete.bodyText' }),
                            )
                          ) {
                            fields.remove(index);
                          }
                        }}>
                        <i className="fa fa-trash" />
                      </Button>
                    </ButtonToolbar>
                  </Col>
                </Row>
              </ListGroupItem>
            ) : null,
          )}
        </ListGroup>
        {fields.length > MAX_FIELDS_DISPLAYED && !showAll && (
          <div className="text-center">
            <Button
              bsStyle="default"
              onClick={() => {
                this.setState({ showAll: true });
              }}>
              <FormattedMessage id="global.more" />
            </Button>
          </div>
        )}
        <Button
          id="add-answer-in-question-button"
          bsStyle="primary"
          className="btn--outline box-content__toolbar"
          onClick={() => {
            dispatch(arrayPush(formName, `${oldMember}.choices`, {}));
            this.setState({
              editIndex: fields.length,
              isCreating: true,
              showQuestionChoiceModal: true,
              editMember: `${oldMember}.choices[${fields.length}]`,
            });
          }}>
          <i className="fa fa-plus-circle" /> <FormattedMessage id="global.add" />
        </Button>
        {type !== 'radio' && (
          <Button
            id="import_choices"
            bsStyle="primary"
            className="btn--outline box-content__toolbar ml-5"
            onClick={() => {
              this.setState({ showModal: true });
            }}>
            <i className="fa fa-upload" /> <FormattedMessage id="import" />
          </Button>
        )}
      </div>
    );
  }
}

const mapStateToProps = (state: GlobalState, props: Props) => {
  const selector = formValueSelector(props.formName);
  return {
    importedResponses: selector(state, `${props.oldMember}.importedResponses`),
    choices: selector(state, `${props.oldMember}.choices`),
  };
};

export default connect<any, any, _, _, _, _>(mapStateToProps)(injectIntl(QuestionChoiceAdminForm));
