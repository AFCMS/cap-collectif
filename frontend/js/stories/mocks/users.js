// @flow

export const author = {
  id: 'user1',
  username: 'admin',
  media: {
    url: 'https://source.unsplash.com/random/150x150',
  },
  _links: {},
  vip: false,
  isViewer: false,
  isAdmin: true,
  isEvaluer: true,
};

export const followers = [
  {
    displayName: 'All Fred',
    isEmailConfirmed: true,
    media: {
      url: 'https://source.unsplash.com/collection/181462',
    },
    _links: {
      profile: 'https://capco.dev',
    },
    contributionsCount: 4,
  },
  {
    displayName: 'guemas',
    isEmailConfirmed: true,
    media: {
      url: 'https://source.unsplash.com/collection/181462',
    },
    _links: {
      profile: 'https://capco.dev',
    },
    contributionsCount: 0,
  },
  {
    displayName: 'Shezeboss',
    isEmailConfirmed: true,
    media: {
      url: 'https://source.unsplash.com/collection/181462',
    },
    _links: {
      profile: 'https://capco.dev',
    },
    contributionsCount: 4,
  },
  {
    displayName: 'AS',
    isEmailConfirmed: true,
    media: {
      url: 'https://source.unsplash.com/collection/181462',
    },
    _links: {
      profile: 'https://capco.dev',
    },
    contributionsCount: 240,
  },
  {
    displayName: 'Louisette',
    isEmailConfirmed: false,
    media: {
      url: 'https://source.unsplash.com/collection/181462',
    },
    _links: {
      profile: 'https://capco.dev',
    },
    contributionsCount: 1,
  },
  {
    contributionsCount: 1,
    isEmailConfirmed: true,
    _links: null,
    media: null,
  },
  {
    displayName: 'guemas',
    isEmailConfirmed: true,
    media: {
      url: 'https://source.unsplash.com/collection/181462',
    },
    _links: {
      profile: 'https://capco.dev',
    },
    contributionsCount: 0,
  },
  {
    displayName: 'Shezeboss',
    isEmailConfirmed: true,
    media: {
      url: 'https://source.unsplash.com/collection/181462',
    },
    _links: {
      profile: 'https://capco.dev',
    },
    contributionsCount: 4,
  },
  {
    displayName: 'AS',
    isEmailConfirmed: true,
    media: {
      url: 'https://source.unsplash.com/collection/181462',
    },
    _links: {
      profile: 'https://capco.dev',
    },
    contributionsCount: 240,
  },
];
