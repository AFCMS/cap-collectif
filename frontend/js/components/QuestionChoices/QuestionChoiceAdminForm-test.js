// @flow
/* eslint-env jest */
import React from 'react';
import { shallow } from 'enzyme';
import { QuestionChoiceAdminForm } from './QuestionChoiceAdminForm';
import { intlMock } from '../../mocks';

const props = {
  dispatch: jest.fn(),
  fields: { length: 1, map: jest.fn(), remove: jest.fn() },
  choices: [
    {
      id: '1',
      title: 'test',
      description: 'thisisnotatest',
      color: 'red',
      image: Object,
    },
  ],
  formName: 'formName',
  oldMember: 'oldMember',
  type: 'type',
  intl: intlMock,
  importedResponses: null,
};

describe('<ProposalUserVoteItem />', () => {
  it('should render correctly', () => {
    const withResponse = {
      ...props,
      importedResponses: {
        data: [{ title: 'response1' }, { title: 'response2' }, { title: 'response3' }],
      },
    };
    const wrapper = shallow(<QuestionChoiceAdminForm {...withResponse} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly modal to import file opened', () => {
    const wrapper = shallow(<QuestionChoiceAdminForm {...props} />);
    wrapper.setState({
      showModal: true,
    });
    expect(wrapper).toMatchSnapshot();
  });
  it('should render correctly modal to edit question', () => {
    const wrapper = shallow(<QuestionChoiceAdminForm {...props} />);
    wrapper.setState({
      editIndex: 3,
      editMember: 'questions[0].choices[3]',
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly without choices', () => {
    const propsNoQuestionChoices = {
      ...props,
      choices: [],
    };
    const wrapper = shallow(<QuestionChoiceAdminForm {...propsNoQuestionChoices} />);
    expect(wrapper).toMatchSnapshot();
  });
});
