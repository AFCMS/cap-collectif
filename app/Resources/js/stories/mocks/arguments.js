// @flow
import { author } from './users';

export const argument = {
  user: author,
  trashedStatus: '',
  createdAt: '1 mars 2016 à 00:00',
  publishedAt: '1 mars 2016 à 00:00',
  body:
    'In placeat reiciendis ut. Officiis praesentium quia minima ut tenetur officiis. Eaque fugit voluptates temporibus suscipit provident culpa culpa. Magni recusandae dolorem aut id.',
  votes: { totalCount: 50 },
};

export const argumentsData = [
  { ...argument, user: { ...argument.user, vip: true } },
  { ...argument },
  { ...argument, trashedStatus: 'INVISIBLE' },
  { ...argument },
];
