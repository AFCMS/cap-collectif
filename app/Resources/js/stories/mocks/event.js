// @flow
import { author } from './users';

export const event = {
  title: 'PHPTourDuTurfu',
  url: '#',
  image: {
    url: 'https://picsum.photos/300/400',
    alt: 'event picture',
  },
  date: {
    start: '2030-03-10 00:00:00',
    end: '2032-06-15 00:00:00',
  },
  location: {
    place: 'Tour Eiffel',
    zip: '75007',
    city: 'Paris',
  },
  category: 'Immobilier',

  user: author,
};
