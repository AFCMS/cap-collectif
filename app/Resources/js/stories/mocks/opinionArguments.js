// @flow
import { author } from './users';

export const opinionArgument = {
  user: author,
  trashedStatus: '',
  createdAt: '1 mars 2016 à 00:00',
  publishedAt: '1 mars 2016 à 00:00',
  body:
    'In placeat reiciendis ut. Officiis praesentium quia minima ut tenetur officiis. Eaque fugit voluptates temporibus suscipit provident culpa culpa. Magni recusandae dolorem aut id.',
  votes: { totalCount: 50 },
};

export const opinionArguments = [
  { ...opinionArgument, user: { ...opinionArgument.user, vip: true } },
  { ...opinionArgument },
  { ...opinionArgument, trashedStatus: 'INVISIBLE' },
  { ...opinionArgument },
];
