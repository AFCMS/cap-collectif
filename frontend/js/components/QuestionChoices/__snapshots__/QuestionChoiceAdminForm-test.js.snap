// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProposalUserVoteItem /> should render correctly 1`] = `
<div
  className="form-group"
  id="questions_choice_panel_personal"
>
  <Connect(injectIntl(QuestionChoiceAdminModal))
    formName="formName"
    isCreating={true}
    member={null}
    onClose={[Function]}
    onSubmit={[Function]}
    show={false}
    type="type"
  />
  <ListGroup
    bsClass="list-group"
  />
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="primary"
    className="btn--outline box-content__toolbar"
    disabled={false}
    id="add-answer-in-question-button"
    onClick={[Function]}
  >
    <i
      className="fa fa-plus-circle"
    />
     
    <FormattedMessage
      id="global.add"
      values={Object {}}
    />
  </Button>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="primary"
    className="btn--outline box-content__toolbar ml-5"
    disabled={false}
    id="import_choices"
    onClick={[Function]}
  >
    <i
      className="fa fa-upload"
    />
     
    <FormattedMessage
      id="import"
      values={Object {}}
    />
  </Button>
</div>
`;

exports[`<ProposalUserVoteItem /> should render correctly modal to edit question 1`] = `
<div
  className="form-group"
  id="questions_choice_panel_personal"
>
  <Connect(injectIntl(QuestionChoiceAdminModal))
    formName="formName"
    isCreating={true}
    member="questions[0].choices[3]"
    onClose={[Function]}
    onSubmit={[Function]}
    show={false}
    type="type"
  />
  <ListGroup
    bsClass="list-group"
  />
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="primary"
    className="btn--outline box-content__toolbar"
    disabled={false}
    id="add-answer-in-question-button"
    onClick={[Function]}
  >
    <i
      className="fa fa-plus-circle"
    />
     
    <FormattedMessage
      id="global.add"
      values={Object {}}
    />
  </Button>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="primary"
    className="btn--outline box-content__toolbar ml-5"
    disabled={false}
    id="import_choices"
    onClick={[Function]}
  >
    <i
      className="fa fa-upload"
    />
     
    <FormattedMessage
      id="import"
      values={Object {}}
    />
  </Button>
</div>
`;

exports[`<ProposalUserVoteItem /> should render correctly modal to import file opened 1`] = `
<div
  className="form-group"
  id="questions_choice_panel_personal"
>
  <Connect(injectIntl(QuestionnaireAdminModalImportResponses))
    choices={
      Array [
        Object {
          "color": "red",
          "description": "thisisnotatest",
          "id": "1",
          "image": [Function],
          "title": "test",
        },
      ]
    }
    dispatch={[MockFunction]}
    fields={
      Object {
        "length": 1,
        "map": [MockFunction] {
          "calls": Array [
            Array [
              [Function],
            ],
            Array [
              [Function],
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
        "remove": [MockFunction],
      }
    }
    formName="formName"
    oldMember="oldMember"
    onAdd={[Function]}
    onClose={[Function]}
    show={true}
    type="type"
  />
  <Connect(injectIntl(QuestionChoiceAdminModal))
    formName="formName"
    isCreating={true}
    member={null}
    onClose={[Function]}
    onSubmit={[Function]}
    show={false}
    type="type"
  />
  <ListGroup
    bsClass="list-group"
  />
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="primary"
    className="btn--outline box-content__toolbar"
    disabled={false}
    id="add-answer-in-question-button"
    onClick={[Function]}
  >
    <i
      className="fa fa-plus-circle"
    />
     
    <FormattedMessage
      id="global.add"
      values={Object {}}
    />
  </Button>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="primary"
    className="btn--outline box-content__toolbar ml-5"
    disabled={false}
    id="import_choices"
    onClick={[Function]}
  >
    <i
      className="fa fa-upload"
    />
     
    <FormattedMessage
      id="import"
      values={Object {}}
    />
  </Button>
</div>
`;

exports[`<ProposalUserVoteItem /> should render correctly without choices 1`] = `
<div
  className="form-group"
  id="questions_choice_panel_personal"
>
  <Connect(injectIntl(QuestionChoiceAdminModal))
    formName="formName"
    isCreating={true}
    member={null}
    onClose={[Function]}
    onSubmit={[Function]}
    show={false}
    type="type"
  />
  <ListGroup
    bsClass="list-group"
  />
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="primary"
    className="btn--outline box-content__toolbar"
    disabled={false}
    id="add-answer-in-question-button"
    onClick={[Function]}
  >
    <i
      className="fa fa-plus-circle"
    />
     
    <FormattedMessage
      id="global.add"
      values={Object {}}
    />
  </Button>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="primary"
    className="btn--outline box-content__toolbar ml-5"
    disabled={false}
    id="import_choices"
    onClick={[Function]}
  >
    <i
      className="fa fa-upload"
    />
     
    <FormattedMessage
      id="import"
      values={Object {}}
    />
  </Button>
</div>
`;
